src/main/java/org/openstreetmap/atlas/geography/atlas/raw/sectioning/WaySectionProcessor.java

```diff
@@ -0,0 +1,906 @@
+                    final Location shapePoint = polyLine.get(index);
+
+                    // Based on a configurable tag filter, a shape point that doesn't intersect
+                    // an edge may also become a node
+                    if (shouldSectionAtLocation(shapePoint))
+                    {
+                        addPointToNodeList(shapePoint, nodesForEdge);
+                    }
+
+                    // TODO - Getting non-intersecting lines from the spatial query results.
+                    // So purposefully specifying "contains shapePoint". Need to resolve this!
+
+                    // If there are other edges intersecting the shape point, it becomes a node
+                    if (locationHasIntersectingLinesMatchingPredicate(shapePoint,
+                            target -> target.getIdentifier() != line.getIdentifier()
+                                    && isAtlasEdge(target)
+                                    && target.asPolyLine().contains(shapePoint)))
+                    {
+                        final Iterable<Point> pointsAtLocation = this.rawAtlas.pointsAt(shapePoint);
+                        if (Iterables.size(pointsAtLocation) > 1)
+                        {
+                            // This shouldn't happen, so let's log and section at all points. By
+                            // definition, all stacked (same location) OSM nodes will be collapsed
+                            // to a single atlas point during raw atlas creation.
+                            logger.error("Detected more than one point at {}", shapePoint);
+                        }
+                        addPointToNodeList(shapePoint, nodesForEdge);
+                    }
+                }
+
+                if (line.isClosed() && nodesForEdge.size() == 0)
+                {
+                    // Handle isolated rings by creating a node at the start and the middle
+                    // shape point, effectively splitting it in half.
+                    addPointToNodeList(polyLine.first(), nodesForEdge);
+                    addPointToNodeList(polyLine.get(polyLine.size() / 2), nodesForEdge);
+                }
+                else if (!line.isClosed())
+                {
+                    // For non-closed loops, the first and last point of the edge are always nodes.
+                    // We don't care about first/last points for rings because these can be
+                    // arbitrary and we only want to section at edge intersections.
+                    addPointToNodeList(polyLine.first(), nodesForEdge);
+                    addPointToNodeList(polyLine.last(), nodesForEdge);
+                }
+
+                // Record the edge and all its nodes
+                changeSet.createEdgeToNodeMapping(line.getIdentifier(), nodesForEdge);
+            }
+            else if (isAtlasArea(line))
+            {
+                changeSet.recordArea(line);
+            }
+            else if (isAtlasLine(line))
```

Comment:
is there a large performance hit for keeping this code around?

Reply:
Good catch! I originally intended to call out lines that didn't qualify as an Edge, Line or Area and handle them in the else block - log and remove from Atlas, but ultimately made the call to have everything be in the Atlas as a Line if it's not an Area or Edge. Looks like I got a little too cute here and making an unnecessary call to check for Line properties.

Reply:
Reconsidering the decision to keep everything as a Line, there are a few cases that make no sense. Examples:
1. Line folding on to itself: https://www.openstreetmap.org/way/320353475#map=19/7.26214/-8.14084
2. JTS country-slicing elements which are points.

Reply:
Updating the area filter to bring in railway and highway platforms and filtering out anything that doesn't fit this definition!

Reply:
awesome! sounds good.